### DISCUSSION TITLE
JSON Error in v2.5.2 and onwards

### DISCUSSION THREAD

**Discussion by BrekiTomasson** [livewire/livewire#3315] [7. Bugs (archived)] [answered]
Was updating a number of Composer requirements today and bumped Livewire up to `v2.5.4`, which immediately throw the old faithful "Unexpected end of JSON input" for all of my Livewire components. Downgraded to `v2.5.3`, same thing. Downgraded to `v2.5.2`, same thing. Downgraded to `v2.5.1`, and it all started working again.

I've been looking through the changes introduced in `v2.5.2` but can't make out what might be causing this. For now, at least, I'm locking my version to `v2.5.1`.

Reading #3252 it looks like this should be fixed in `v2.5.3`, but that is not what I am experiencing - every published version after `v2.5.2` is broken for me.

**Comment by joshhanley** [comment, marked as answer]
@BrekiTomasson a fix has been tagged for this, please upgrade to Livewire 2.5.5 and let us know if you are still having any issues 🙂

There was multiple issues that resulted in the same error in the console (although they were unrelated). This particular one was due to a patching of `setAttribute` in 2.5.2 to support the `@` symbol, for things like `@click` when using Alpine. But it triggered issues for normal set attribute, causing breakages with other JS libraries.

**Reply by BrekiTomasson** [reply]
Yeah, `v2.5.5` seems to have fixed it! Thanks!
